packages/playwright-core/src/remote/playwrightConnection.ts

```diff
@@ -104,47 +102,41 @@ export class PlaywrightConnection {
       return;
     }
 
-    this._root = new RootDispatcher(this._dispatcherConnection, async (scope, options) => {
+    this._root = new RootDispatcher(this._dispatcherConnection, async scope => {
       await startProfiling();
       if (clientType === 'reuse-browser')
         return await this._initReuseBrowsersMode(scope);
       if (clientType === 'pre-launched-browser-or-android')
         return this._preLaunched.browser ? await this._initPreLaunchedBrowserMode(scope) : await this._initPreLaunchedAndroidMode(scope);
       if (clientType === 'launch-browser')
-        return await this._initLaunchBrowserMode(scope, options);
+        return await this._initLaunchBrowserMode(scope);
       throw new Error('Unsupported client type: ' + clientType);
     });
   }
 
-  private async _initLaunchBrowserMode(scope: RootDispatcher, options: channels.RootInitializeParams) {
+  private async _initLaunchBrowserMode(scope: RootDispatcher) {
     debugLogger.log('server', `[${this._id}] engaged launch mode for "${this._options.browserName}"`);
-    const playwright = createPlaywright({ sdkLanguage: options.sdkLanguage, isServer: true });
-
-    const ownedSocksProxy = await this._createOwnedSocksProxy(playwright);
+    const ownedSocksProxy = await this._createOwnedSocksProxy(this._playwright);
     let browserName = this._options.browserName;
     if ('bidi' === browserName) {
       if (this._options.launchOptions?.channel?.toLocaleLowerCase().includes('firefox'))
         browserName = 'bidiFirefox';
       else
         browserName = 'bidiChromium';
     }
-    const browser = await playwright[browserName as 'chromium'].launch(serverSideCallMetadata(), this._options.launchOptions);
+    const browser = await this._playwright[browserName as 'chromium'].launch(serverSideCallMetadata(), this._options.launchOptions);
 
-    this._cleanups.push(async () => {
-      for (const browser of playwright.allBrowsers())
-        await browser.close({ reason: 'Connection terminated' });
-    });
+    this._cleanups.push(() => browser.close({ reason: 'Connection terminated' }));
```

Comment:
While this change here makes sense, let's keep the "close all browsers upon disconnect" logic in `PlaywrightServer` when we are running in non-extension mode for now.